### ISSUE TITLE
std::runtime_error -- what():  qds_device::wait() unexpected command state

### ISSUE THREAD

**Issue by hecmay** [Xilinx/mlir-aie#1751]
I was running cascade matrix multiple example from programming examples: https://github.com/Xilinx/mlir-aie/tree/main/programming_examples/basic/matrix_multiplication/cascade

I noticed that the code only works when `M` or `m` is greater than 64. When I set them to smaller values, say `M = 32, n = 32`, the host code will throw the following error:

```bash
Running Kernel (iteration 0).
terminate called after throwing an instance of 'std::runtime_error'
  what():  qds_device::wait() unexpected command state
Aborted (core dumped)
make: *** [/home/ubuntu/mlir-aie-test/programming_examples/basic/matrix_multiplication/cascade/../makefile-common:114: run] Error 134
```

No error from XCLBIN compilation process, and it seems to be something wrong with runtime? Any idea how this can be fixed?

**Comment by hecmay** [comment]
Running into exactly the same error in matrix vector sample code if the `n_cores` is > 1: https://github.com/Xilinx/mlir-aie/blob/main/programming_examples/basic/matrix_multiplication/matrix_vector/aie2.py#L21

**Comment by hecmay** [comment]
I am not sure if this error is caused by my environment setup. I am using [Phoenix Point Mini PC: Minisforum UM790 Pro : AMD Ryzen™ 9 7940HS](https://store.minisforum.com/products/minisforum-um790-pro?variant=43865372786933). I followed every single step in this README: https://github.com/Xilinx/mlir-aie/blob/main/docs/buildHostLin.md

```bash
Linux kernel: 6.10
Vitis 2023.2
AMDXDNA: 2.18.0_20240825, 537a509a3ab1b698c9c9f6ebcd88035b2fe8359b
```

Can anyone reproduce the issue? Any help would be highly appreciated. Thanks! @stephenneuendorffer @fifield @hunhoffe @Yu-Zhewen @makslevental

**Comment by PisonJay** [comment]
Me too. Getting same error with Ryzen AI 9 365.

**Comment by PisonJay** [comment]
I guess that current infrastructure only supports XDNA1 (AIE2) architecture. Strix Point, XDNA2 (with code name AIEP) is not supported yet. So does [Peano], listing XNDA2 as "coming soon". The only usable runtime is ONNX runtime from [Ryzen AI SDK](https://ryzenai.docs.amd.com/en/latest/inst.html), only available on Windows at current time.

**Comment by hecmay** [comment]
> I guess that current infrastructure only supports XDNA1 (AIE2) architecture. Strix Point, XDNA2 (with code name AIEP) is not supported yet. So does [Peano](https://github.com/Xilinx/llvm-aie/blob/aie-public/README.md?plain=1#L8), listing XNDA2 as "coming soon". The only usable runtime is ONNX runtime from [Ryzen AI SDK](https://ryzenai.docs.amd.com/en/latest/inst.html), only available on Windows at current time.

Not so sure if that's the reason, as on my side, the program is still runnable in some cases if the M/N/K values make the runtime happy. 

And I do not think ONNX runtime is used in these examples from this repo. It should be some weird problems from Xilinx Runtime: https://github.com/Xilinx/mlir-aie/blob/main/programming_examples/basic/matrix_multiplication/test.cpp#L23-L25

**Comment by aFuerst** [comment]
Any change/progress with this?
I have been trying to run a variety of the examples and they all fail with this identical error

On a new AMD Ryzen AI 9 HX 370 w/ Radeon 890M
Build setup from `xrt-smi examine`
```
System Configuration
  OS Name              : Linux
  Release              : 6.14.0-rc2+
  Machine              : x86_64
  CPU Cores            : 24
  Memory               : 27667 MB
  Distribution         : Ubuntu 24.04.2 LTS
  GLIBC                : 2.39
  Model                : SER9
  BIOS Vendor          : American Megatrends International, LLC.
  BIOS Version         : STX.3xx.SER9.V103.P8C0M0C15.73.Link.SCPC

XRT
  Version              : 2.19.0
  Branch               : HEAD
  Hash                 : f02de266f785111094b02d32a9012bd0da3abd58
  Hash Date            : 2025-02-20 09:23:11
  amdxdna              : 2.19.0_20250306, f3ce30a590714e786e57a003f446d617dc67f7ef
  virtio-pci           : unknown, unknown
  NPU Firmware Version : 1.0.4.73

Device(s) Present
|BDF             |Name          |
|----------------|--------------|
|[0000:65:00.1]  |RyzenAI-npu4  |
```

**Comment by jgmelber** [comment]
Are you building your designs with the device NPU2? `npu2==npu4`

**Comment by aFuerst** [comment]
Not not specifying building anything with a specific device, npu2 or otherwise.
None of the examples I've tried have had that option either. Such as `ml/conv2d_fused_relu` or `basic/matrix_multiplication`.

Are you thinking it's because my device isn't supported yet?


I'm building `xrt` by passing `-npu`, which I know is different though.

**Comment by jgmelber** [comment]
> Not not specifying building anything with a specific device, npu2 or otherwise.
> None of the examples I've tried have had that option either. Such as `ml/conv2d_fused_relu` or `basic/matrix_multiplication`.

Yes you are using a Strix NPU, so for any example you need to target NPU2 as the device. Many examples are supported, GEMM is a WIP. You can see PR #2067 for GEMM and the issue tracking which programming examples are left to port for NPU2 #2092. 

Can you please try the `passthrough_kernel` example as a smoke test?
